src/feature/feature-pose.cpp

```diff
@@ -25,327 +25,17 @@
-  return X * faNufafbDes;
-}
-template <>
-Vector6d convertVelocity<R3xSO3_t>(const MatrixHomogeneous &M,
-                                   const MatrixHomogeneous &Mdes,
-                                   const Vector &faNufafbDes) {
-  Vector6d nu;
-  nu.head<3>() =
-      faNufafbDes.head<3>() - M.translation().cross(faNufafbDes.tail<3>());
-  nu.tail<3>() = Mdes.rotation().transpose() * faNufafbDes.tail<3>();
-  return nu;
-}
-
-template <Representation_t representation>
-Vector &FeaturePose<representation>::computeErrorDot(Vector &errordot,
-                                                     int time) {
-  typedef typename LG_t<representation>::type LieGroup_t;
-  check(*this);
-
-  errordot.resize(dimensionSOUT(time));
-  const Flags &fl = selectionSIN(time);
-  if (!faNufafbDes.isPlugged()) {
-    errordot.setZero();
-    return errordot;
-  }
-
-  q_faMfb.recompute(time);
-  q_faMfbDes.recompute(time);
-  faNufafbDes.recompute(time);
-
-  const MatrixHomogeneous &_faMfbDes = faMfbDes(time);
-
-  Eigen::Matrix<double, 6, 6, Eigen::RowMajor> Jminus;
-
-  LieGroup_t().template dDifference<pinocchio::ARG0>(
-      q_faMfbDes.accessCopy(), q_faMfb.accessCopy(), Jminus);
-  Vector6d nu = convertVelocity<LieGroup_t>(faMfb(time), _faMfbDes,
-                                            faNufafbDes.accessCopy());
-  unsigned int cursor = 0;
-  for (unsigned int i = 0; i < 6; ++i)
-    if (fl(i))
-      errordot(cursor++) = Jminus.row(i) * nu;
-
-  return errordot;
-}
-
-/* Modify the value of the reference (sdes) so that it corresponds
- * to the current position. The effect on the servo is to maintain the
- * current position and correct any drift. */
-template <Representation_t representation>
-void FeaturePose<representation>::servoCurrentPosition(const int &time) {
-  check(*this);
-
-  const MatrixHomogeneous &_oMja = (oMja.isPlugged() ? oMja.access(time) : Id),
-                          _jaMfa =
-                              (jaMfa.isPlugged() ? jaMfa.access(time) : Id),
-                          _oMjb = oMjb.access(time),
-                          _jbMfb =
-                              (jbMfb.isPlugged() ? jbMfb.access(time) : Id);
-  faMfbDes = (_oMja * _jaMfa).inverse(Eigen::Affine) * _oMjb * _jbMfb;
-}
-
-static const char *featureNames[] = {"X ", "Y ", "Z ", "RX", "RY", "RZ"};
-template <Representation_t representation>
-void FeaturePose<representation>::display(std::ostream &os) const {
-  os << CLASS_NAME << "<" << name << ">: (";
-
-  try {
-    const Flags &fl = selectionSIN.accessCopy();
-    bool first = true;
-    for (int i = 0; i < 6; ++i)
-      if (fl(i)) {
-        if (first) {
-          first = false;
-        } else {
-          os << ",";
-        }
-        os << featureNames[i];
-      }
-    os << ") ";
-  } catch (ExceptionAbstract e) {
-    os << " selectSIN not set.";
-  }
-}
+template class FeaturePose<R3xSO3Representation>;
+template class FeaturePose<SE3Representation>;
```

Comment:
It seems this should be inside
```cpp
namespace dynamicgraph {
namespace sot {
}
}
```
(so that we can remove the `using namespace ...` line).